### DISCUSSION TITLE
Error when evaluating 'strategy' for job 'Manual_Deployment'. Expected type: Sequence.

### DISCUSSION THREAD

**Discussion by bturner2** [community/community#42131] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi all, this is my first discussion post.

I am trying to create a rolling deployment by using Workflow Strategies and Matrices. 

Currently I:

- Obtain Instance IDs in a previous step. I set these as a env variable and set them as an output.
`        instances=[$(aws elb describe-load-balancers --load-balancer-names <lb name> | jq -r .LoadBalancerDescriptions[].Instances[].InstanceId | paste -sd , -)]
        echo "INSTANCE_ID=$instances" >> $GITHUB_ENV`
`    outputs: 
      instanceIDs : ${{ env.INSTANCE_ID }}`
- The output looks like this:
`[i-xxxxxxxxx
,i-xxxxxxxxx]`
- I then attempt to use these instance IDs as a variable in my matrix in the next job:
`  Manual_Deployment: `
   ` name: 'Rolling Deployment'`
   ` runs-on: windows-latest`
   ` if: ${{ github.event_name == 'workflow_dispatch' }}`
    `needs: <previous-job>`
    `strategy:`
     ` max-parallel: 1`
    `  matrix:`
     `   instance-ids: ${{ needs.status_check.outputs.instanceIDs }}`

Error when evaluating 'strategy' for job 'Manual_Deployment'. : Unexpected type of value '[i-xxxxxxx,i-xxxxxxx]', expected type: Sequence.

Can anyone help me understand how I should be formatting this array for github to accept it?

**Comment by airtower-luna** [comment, marked as answer]
The answer has two parts:

1. Format the output data as JSON, e.g. `["i-xxxxxxxxx", "i-xxxxxxxxx"]`
2. You can then turn the JSON string into an actual sequence using [`fromJSON()`](https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson), e.g. `instance-ids: ${{ fromJSON(needs.status_check.outputs.instanceIDs) }}`

**Reply by Mykp3** [reply]
Thanks lad) Much appreciated!
